elasticsearch-core/src/main/scala/com/sumologic/elasticsearch/restlastic/RestlasticSearchClient2.scala

```diff
@@ -35,9 +35,9 @@ class RestlasticSearchClient2(endpointProvider: EndpointProvider, signer: Option
 
   override val version = V2
 
-  def deleteByQuery(index: Index, tpe: Type, deleteQuery: QueryRoot, waitForCompletion: Boolean): Future[RawJsonResponse] = {
+  def deleteByQuery(index: Index, tpe: Type, deleteQuery: QueryRoot, waitForCompletion: Boolean): Future[DeleteByQuerySearchResponse] = {
     implicit val ec = indexExecutionCtx
-    deleteDocuments(index, tpe, deleteQuery).map(resp => RawJsonResponse(resp.toString()))
+    deleteDocuments(index, tpe, deleteQuery).map(resp => DeleteByQuerySearchResponse(resp.size))
```

Comment:
I think it would be better to return `-1` when `waitForCompletion` is set to `false`. 
```
deleteDocuments(index, tpe, deleteQuery).map(_ => DeleteByQuerySearchResponse(-1))
```
This will make the ES 2.3 and ES 6 tests consistent.

Reply:
Done